input/otlp/traces.go

```diff
@@ -365,39 +436,43 @@ func TranslateTransaction(
 
 	switch foundSpanType {
 	case httpSpan:
-		event.HTTP = http
+		if !proto.Equal(&http, &modelpb.HTTP{}) {
+			event.Http = &http
+		}
```

Comment:
If `foundSpanType == httpSpan`, shouldn't we know for a fact that http is non-empty?

Reply:
Removed :+1:

Reply:
It seems that's not the case. See the failing test:

```
 === RUN   TestConsumer_JaegerSpan/jaeger_https_default_port
    traces_test.go:1224:   map[string]any{
          	"events": []any{
          		map[string]any{
          			... // 3 identical entries
          			"event":     map[string]any{"duration": float64(7.9e+10), "outcome": string("unknown")},
          			"host":      map[string]any{"hostname": string("host-abc")},
        + 			"http":      map[string]any{},
          			"parent":    map[string]any{"id": string("0000000058585858")},
          			"processor": map[string]any{"event": string("span"), "name": string("transaction")},
          			... // 5 identical entries
          		},
          	},
          }
```

Reply:
Hmm true, because you might have just `http.url`, which translates to `url.*` fields in our model. Let's revisit this later to see if we can find a way to avoid this `proto.Equal` call.